### PR TITLE
feat(rollup): support EIP-1559 in chunk/batch encoding/decoding

### PR DESCRIPTION
**Pull request by colinlyguo**
### Purpose or design rationale of this PR

This PR can be deployed before `l2geth`'s upgrade.

This PR uses l2geth's built-in transaction JSON encoding/decoding:
1. The `TransactionData` structure is moved from the `l2geth` repository to this repository, ensuring backward compatibility.
2. Adding a new column named `transactions_rlp` to store transactions' rlp encoding.

Additionally, this PR removes redundant `block/chunk/batch` encoding/decoding implementation, using the same implementation in `l2geth` repository.

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] feat: A new feature

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206327371641352

### CODE REVIEW

File: common/types/transaction_data.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+				GasPrice: oldTx.GasPrice.ToInt(),
+				Data:     data,
+				V:        oldTx.V.ToInt(),
+				R:        oldTx.R.ToInt(),
+				S:        oldTx.S.ToInt(),
+			})
+			txs = append(txs, newTx)
+
+		case types.L1MessageTxType:
+			newTx := types.NewTx(&types.L1MessageTx{
```
Comment (colinlyguo) [inline common/types/transaction_data.go:89]: the `Sender` field is lost here, thus cannot recover L1 message transaction's hash used in the chunk's decoding. @Thegaram

Comment (colinlyguo) [reply common/types/transaction_data.go:89]: new code:
```
	// concatenate l1 and l2 tx hashes
	for _, block := range c.Blocks {
		var l1TxHashes []byte
		var l2TxHashes []byte
		for _, tx := range block.Transactions {
			if tx.Type() == types.L1MessageTxType {
				l1TxHashes = append(l1TxHashes, tx.Hash().Bytes()...)
			} else {
				l2TxHashes = append(l2TxHashes, tx.Hash().Bytes()...)
			}
		}
		dataBytes = append(dataBytes, l1TxHashes...)
		dataBytes = append(dataBytes, l2TxHashes...)
	}
```

**Comment by colinlyguo** [comment]
The coverage decrease of unit tests in `/common` is caused by removing code (using l2geth's implementation).

**Comment by bzarka** [comment]
Hi,

Based on the provided information, the purpose or design rationale of this pull request (PR) appears to be:

1. **Utilization of l2geth's built-in transaction JSON encoding/decoding**: The PR moves the TransactionData structure from the l2geth repository to the current repository to ensure backward compatibility. It also adds a new column named `transactions_rlp` to store transactions' RLP encoding.

2. **Reduction of redundancy**: The PR removes redundant block/chunk/batch encoding/decoding implementation by utilizing the existing implementation in the l2geth repository.

3. **Deployment timing**: The PR is intended to be deployed before l2geth's upgrade, indicating a focus on timely integration and compatibility with upcoming changes.

The PR title should follow conventional commits to align with the project's commit message convention for squash merging. Depending on the specific changes made, the PR title might start with one of the following types:

- feat: for new features or functionalities added
- fix: for bug fixes or corrections
- refactor: for code refactoring without changing its external behavior
- chore: for changes related to build processes, tooling, or documentation
- docs: for documentation updates
- style: for code style changes that do not affect the code's functionality

The actual PR title should provide a concise summary of the changes made and may include additional information relevant to the context of the PR.

**Comment by colinlyguo** [comment]
Same reason as https://github.com/scroll-tech/go-ethereum/pull/622.

**Pull request closed without merging**